server/database/handle-goal-data.js

```diff
@@ -1,41 +1,63 @@
 const dbClient = require('./db_connection.js');
 
-const getGoal = `
-  SELECT *
-    from goals
-  WHERE goal_id = $1
-`;
-
-const addGoal = `
-  INSERT into goals (goal_id, user_id, title, icon, date_created)
-  VALUES ($1, $2, $3, $4, $5)
-  RETURNING goal_id
-`;
+/* queries */
+const queries = require('./queries.js');
+const addRatings = require('./add-ratings.js');
+
+/* helper functions */
+const { getNewGoalData, findNewRatings, } = require('./../helpers/handle-goals.js');
+
+// compares new goal from client with old goal from DB, adds missing ratings to database
+const updateGoal = (dbGoal, clientGoal, callback) => {
+
+  // check for new ratings
+  dbClient.query(queries.getRatings, [ clientGoal.id, ], (dbRatingsErr, dbRatingsRes) => {
+
+    if (dbRatingsErr) {
+      return callback('error getting ratings from database');
+    }
+
+    const dbRatings = dbRatingsRes.rows;
+
+    if (dbRatings.length >= clientGoal.ratings.length) {
+      return callback(null, clientGoal.id);
+    }
+
+    const newRatings = findNewRatings(dbRatings, clientGoal.ratings);
+
+    addRatings(newRatings, dbGoal, callback);
+  });
+};
 
 module.exports = (goal, user_id, callback) => {
+
   const goalId = goal.id;
 
-  dbClient.query(getGoal, [ goalId, ], (getGoalErr, getGoalResult) => {
+  dbClient.query(queries.getGoal, [ goalId, ], (getGoalErr, getGoalResult) => {
     if (getGoalErr) {
-      console.log("Error retreiving goal from database, ", getGoalErr);
-      return callback('error data from database');
+      console.log(getGoalErr);
+      return callback('error getting data from database');
     }
 
     if (getGoalResult.rows.length) {
-      // adapt this for new ratings / when we have editing and deleting functionality
-      return callback(null, { alreadyExists : true, });
+      // goal already exists, check for new ratings
+      // adapt this when we have editing and deleting functionality
```

Comment:
Can we update this comment. Ie I think it would be good to explicitly outline the problems with this approach and give a loose description of what needs to be done.

I think I am right in saying fundamentally, this approach will fail in the future that I would be good to make that clear.

Do you agree?